lib/ash/actions/read/read.ex

```diff
@@ -855,7 +855,7 @@ defmodule Ash.Actions.Read do
 
     if missing_pkeys? ||
          (Enum.empty?(must_be_reselected) && Enum.empty?(query.aggregates) &&
-            Enum.empty?(calculations_in_query)) do
+            Enum.empty?(calculations_in_query) && Enum.empty?(query.load)) do
```

Comment:
I think this check may be problematic. Specifically, the "load" statement should already be "flushed out" to `aggregates` and `calculations_in_query`. What this would say to be is that the work to `add_relationship_count_aggregates/1` needs to be done earlier in the pipeline, before we split off the aggregates/calcs, which would allow us to leave this section of code unchanged.

Reply:
I followed your advice and moved the change before in the pipeline.
However I am bit lost with what's going on in the code, so maybe double check my guess :)